static/js/publisher/release/releasesController.js

```diff
@@ -17,14 +18,19 @@ import {
   getReleaseDataFromChannelMap
 } from "./releasesState";
 
-export default class ReleasesController extends Component {
+class ReleasesController extends Component {
   constructor(props) {
     super(props);
 
     // init channel data in revisions list
+    // TODO: should be done in reducers?
     const revisionsMap = getRevisionsMap(this.props.releasesData.revisions);
```

Comment:
I think (if I understand this correctly) I'd do this in the `mapStateToProps`

Reply:
It's basically transforming an array of revisions into a map (keyed by revision id), so I think reducer could do it (especially that we need to do it on successful release fetch as well). Similarly - updating list of channels for each revisions.

Reply:
We've been using reselect in BSI as well. Currently I have a bunch of non-memoized selector functions (https://github.com/canonical-websites/snapcraft.io/blob/master/static/js/publisher/release/selectors/index.js), but have in plans to use reselect to memoize them.